llvm/lib/Target/AArch64/AArch64SelectionDAGInfo.cpp

```diff
@@ -76,15 +82,85 @@ SDValue AArch64SelectionDAGInfo::EmitMOPS(AArch64ISD::NodeType SDOpcode,
   }
 }
 
+SDValue AArch64SelectionDAGInfo::EmitSpecializedLibcall(
```

Comment:
I see how you changed the handling of different return types, but we still have the code for the 'size' that is uniform for the three case-statements at the end, but which isn't generic for other libcalls (if we ever add them). It's probably better to just rename this function to what it's meant to be: `EmitStreamingCompatibleMemLibCall` and then fix `RetTy` to be `PointerType::getUnqual()`, rather than setting it in each case statement.

Reply:
Done.